src/App.tsx

```diff
@@ -1,14 +1,55 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { StatusFilterValue, Todo } from './types/Todo';
+import { getTodos } from './api';
+import { TodoModal } from './components/TodoModal';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [statusFilter, setStatusFilter] = useState<StatusFilterValue>('all');
+  const [query, setQuery] = useState('');
+
+  const handleTogglingTodo = (todo: Todo | null) => {
+    setSelectedTodo(todo);
+  };
+
+  useEffect(() => {
+    setLoading(true);
+    getTodos()
+      .then(setTodos)
+      .finally(() => setLoading(false));
+  }, []);
+
+  const getPreparedTodos = () => {
+    let result = [...todos];
+
+    if (statusFilter === 'completed') {
+      result = result.filter(todo => todo.completed);
+    }
+
+    if (statusFilter === 'active') {
+      result = result.filter(todo => !todo.completed);
+    }
```

Comment:
- Move this function outside the component
- You don't need to create a copy as `filter` doesn't mutate an array
- Use switch case and enum instead of `if` and magic values